src/Provider/AbstractProvider.php

```diff
@@ -702,18 +702,26 @@ protected function parseResponse(ResponseInterface $response)
      * Custom mapping of expiration, etc should be done here. Always call the
      * parent method when overloading this method.
      *
-     * @param  mixed $result
-     * @return array
+     * @param  array|string $response
+     * @return array|string
+     * @throws UnexpectedValueException
      */
-    protected function prepareAccessTokenResponse(array $result)
+    protected function prepareAccessTokenResponse($response)
```

Comment:
> Should parseResponse() or checkResponse() always ensure the value is an array before passing it back up the chain?

I don't think so, as they could both be used outside of the access token request (I think).